Utils/azuretools-core/src/com/microsoft/azuretools/sdkmanage/AzureCliAzureManager.java

```diff
@@ -22,193 +22,112 @@
+        final OffsetDateTime now = LocalDateTime.now().atZone(ZoneId.systemDefault()).toOffsetDateTime().withOffsetSameInstant(ZoneOffset.UTC);
+        if (token.second().isBefore(now)) {
+            token = this.getAccessTokenViaCli(tid);
+            tenantTokens.put(tid, token);
         }
-        final Azure azure = auth().withSubscription(sid);
-        // TODO: remove this call after Azure SDK properly implements handling of unregistered provider namespaces
-        AzureRegisterProviderNamespaces.registerAzureNamespaces(azure);
-        sidToAzureMap.put(sid, azure);
-        return azure;
-    }
-
-    @Override
-    public AppPlatformManager getAzureSpringCloudClient(String sid) {
-        return isSignedIn() ? sidToAzureSpringCloudManagerMap.computeIfAbsent(sid, s ->
-                buildAzureManager(AppPlatformManager.configure()).authenticate(azureCliCredentials, s)) : null;
-    }
-
-    @Override
-    public InsightsManager getInsightsManager(String sid) {
-        return isSignedIn() ? sidToInsightsManagerMap.computeIfAbsent(sid, s ->
-                buildAzureManager(InsightsManager.configure()).authenticate(azureCliCredentials, s)) : null;
-    }
-
-    @Override
-    public List<Subscription> getSubscriptions() {
-        return isSignedIn() ? authenticated.subscriptions().list() : Collections.EMPTY_LIST;
-    }
-
-    @Override
-    public List<Pair<Subscription, Tenant>> getSubscriptionsWithTenant() {
-        if (!isSignedIn()) {
-            return Collections.EMPTY_LIST;
-        }
-        final Tenant subscriptionTenant = authenticated.tenants().list().stream()
-                .filter(tenant -> StringUtils.equals(tenant.tenantId(), authenticated.tenantId()))
-                .findFirst().orElse(null);
-        if (subscriptionTenant == null) {
-            return Collections.EMPTY_LIST;
-        }
-        final List<Pair<Subscription, Tenant>> result = new ArrayList<>();
-        for (Subscription subscription : getSubscriptions()) {
-            result.add(new Pair<>(subscription, subscriptionTenant));
-        }
-        return result;
-    }
-
-    @Override
-    public Settings getSettings() {
-        return settings;
-    }
-
-    @Override
-    public SubscriptionManager getSubscriptionManager() {
-        return subscriptionManager;
-    }
-
-    @Override
-    public void drop() throws IOException {
-        authenticated = null;
-        azureCliCredentials = null;
-        environment = null;
-        subscriptionManager.cleanSubscriptions();
+        return token.first();
     }
 
-    @Override
-    public KeyVaultClient getKeyVaultClient(String tid) {
-        if (!isSignedIn()) {
-            return null;
+    /**
+     * refer https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/identity/azure-identity/src/main/java/com/azure/
+     *       identity/implementation/IdentityClient.java#L366
+     */
+    private Pair<String, OffsetDateTime> getAccessTokenViaCli(String tid) {
+        //
```

Comment:
remove empty //

Reply:
check id for valid guid

Reply:
do we have to check if the `tid` is valid GUID? is `tid` certainly a valid GUID?